core/src/avm1/object/xml_node_object.rs

```diff
@@ -79,10 +59,12 @@ impl<'gc> TObject<'gc> for XmlNodeObject<'gc> {
         activation: &mut Activation<'_, 'gc, '_>,
         this: Object<'gc>,
     ) -> Result<Object<'gc>, Error<'gc>> {
-        Ok(XmlNodeObject::empty_node(
+        Ok(Self::from_xml_node(
             activation.context.gc_context,
+            XmlNode::new_text(activation.context.gc_context, AvmString::default()),
```

Comment:
(tbh the old `empty_node` utility sounds better than having to manually make text nodes, IMO)

Reply:
Agree, but I plan to make a PR that greatly simplifies node types.